src/com/amazon/ion/benchmark/WriteRandomIonValues.java

```diff
@@ -698,4 +642,158 @@ public static void constructLobs(Random random, IonType type, IonWriter writer)
+                writer.setFieldName(fieldName);
+                IonType type = null;
+                type = IonType.valueOf(value.get("type").toString().toUpperCase());
+                switch (type) {
+                    // if more types of Ion data are available, the logic should be added below.
+                    case STRING:
+                        String codepointRange = "[0, 1114111]";
+                        List<Integer> range = parseRange(codepointRange);
+                        int codePointsLengthBound = 20;
+                        if (value.get("codepoint_length") != null) {
+                            codePointsLengthBound = WriteRandomIonValues.parseConstraints(value, "codepoint_length");
+                        }
+                        String stringValue = WriteRandomIonValues.constructString(range, codePointsLengthBound);
+                        writer.writeString(stringValue);
+                        break;
+                    case TIMESTAMP:
+                        // call the function that extract the precision of the timestamp
+                        Timestamp.Precision[] precisions = Timestamp.Precision.values();
+                        Timestamp.Precision precision = precisions[random.nextInt(precisions.length)];
+                        if (value.get("timestamp_precision") != null) {
+                            precision = WriteRandomIonValues.getTimestampPrecisionTemplate(value);
+                        }
+                        Timestamp timestampValue = WriteRandomIonValues.writeTimestamp(precision, null);
+                        writer.writeTimestamp(timestampValue);
+                        break;
+                }
+            }
+            writer.stepOut();
+        }
+    }
+
+    /**
+     * Extract the value of the constraints, select from the set (occurs | container_length | codepoint_length).
+     * @param value is the Ion struct which contain the current constraint field
+     * @param keyWord is the field name of the constraint
+     * @return the value of the current constraint.
+     * @throws IOException if an error occur when constructing the IonReader.
+     */
+    public static int parseConstraints(IonStruct value, String keyWord) throws IOException {
+        Random random = new Random();
+        int result = 0;
+        try (IonReader reader = IonReaderBuilder.standard().build(value)) {
+            reader.next();
+            reader.stepIn();
+            while (reader.next() != null) {
+                if (reader.getFieldName().equals(keyWord)) {
+                    IonType type = reader.getType();
+                    switch (type) {
+                        case INT:
+                            result = reader.intValue();
+                            break;
+                        case SYMBOL:
+                            if (reader.stringValue().equals("required")) {
+                                result = 1;
+                            } else {
```

Comment:
I recommend making the code more explicit by making this an `else if` that checks for "optional". Then your `else` branch can throw an `IllegalArgumentException` or something because it would mean that the schema contained something not supported.

Reply:
> I recommend making the code more explicit by making this an `else if` that checks for "optional". Then your `else` branch can throw an `IllegalArgumentException` or something because it would mean that the schema contained something not supported.

Sounds great! Change has been updated.